### DISCUSSION TITLE
I'm trying to stream from obs to Ant Media Server - it gives error: Unable to access the specified channel or stream key, check your stream key. #5749

### DISCUSSION THREAD

**Discussion by mekya** [ant-media/Ant-Media-Server#5751] [General]
Question from @FranciscoNetoBr moved from https://github.com/ant-media/Ant-Media-Server/issues/5749 to here

> I'm trying to stream from obs to Ant Media Server - rtmp://144.22.195.199/WebRTCApp/HtjZIteFGUE11697042289657 , it gives error: Unable to access the specified channel or stream key, check your stream key. If this is correct, there may be a problem connecting to the server.
> 
> I've already checked the ports, etc. I use an Oracle VPN and followed the procedure https://www.minextuts.com/create-24-7-live-streaming-server-free-multistream-enabled/ for configuration.
> 
> I did a lot of research before posting here, I hope I can solve the problem.
> 
> Thanks in advance.

**Comment by FranciscoNetoBr** [comment]
Can someone help me? I still haven't been able to find a solution.

**Reply by muratugureminoglu** [reply]
Hi @FranciscoNetoBr 

Can you please share with me the instance type ? Because I m getting the following error

```
[rtmp @ 0x557b632450c0] Server error: current system resources not enough
rtmp://144.22.195.199/WebRTCApp/stream1: Operation not permitted
```

What is the output of the following command ? 

```
ps aux --sort=+pcpu | awk '{sum+=$3} END {print sum}'
```

**Reply by drelrichy** [reply]
i got 8.1

**Comment by FranciscoNetoBr** [comment]
Thank you for your attention... Follow the feedback...

]ps aux --sort=+pcpu | awk '{sum+=$3} END {print sum}'
Comando ']ps' não encontrado, você quis dizer:
  comando 'pps' do deb libpmix-bin (4.1.2-2ubuntu1)
  comando 'ps' do deb procps (2:3.3.17-6ubuntu2)
  comando 'qps' do deb qps (2.3.0-0ubuntu1)
  comando 'ips' do deb ips (4.0-1.1)
  comando 'bps' do deb bpfcc-introspection (0.18.0+ds-2)
  comando 'jps' do deb openjdk-11-jdk-headless (11.0.20.1+1-0ubuntu1~22.04)
  comando 'jps' do deb openjdk-17-jdk-headless (17.0.8.1+1~us1-0ubuntu1~22.04)
  comando 'jps' do deb openjdk-18-jdk-headless (18.0.2+9-2~22.04)
  comando 'jps' do deb openjdk-19-jdk-headless (19.0.2+7-0ubuntu3~22.04)
  comando 'jps' do deb openjdk-8-jdk-headless (8u382-ga-1~22.04.1)
Experimente: sudo apt install <deb name>

**Reply by muratugureminoglu** [reply]
The correct command is as follows, Please pay attention, not `]ps`, also please provide the instance type. 

```
ps aux --sort=+pcpu | awk '{sum+=$3} END {print sum}'
```

**Reply by FranciscoNetoBr** [reply]
root@:~# ps aux --sort=+pcpu | awk '{sum+=$3} END {print sum}'
362.8

**Reply by FranciscoNetoBr** [reply]
Would this be the expected return?

**Reply by muratugureminoglu** [reply]
@FranciscoNetoBr 

Since your system is running currently under load, you cannot broadcast over RTMP in any way. You should investigate why the system is under high load and then solve it. I also think that the instance type you used is insufficient. Select a compute-based instance (with at least 4vCore), then try again.

https://docs.oracle.com/en-us/iaas/Content/Compute/References/computeshapes.htm#vmshapes__vm-optimized

**Reply by FranciscoNetoBr** [reply]
Okay, I'll solve this high-processing situation and see what I can come up with.

**Comment by FranciscoNetoBr** [comment]
I tried again on a fully dedicated server, and as you can see in the screenshots below.

rtmp://129.148.27.211/LiveApp/Z4Ub86CA2QptgLoE1698735366640
ps aux --sort=+pcpu | awk '{sum+=$3} END {print sum}'
1.7
![image](https://github.com/ant-media/Ant-Media-Server/assets/127934645/d3d9fdb5-dd03-4d0a-b1ed-d97a578090b1)
![image](https://github.com/ant-media/Ant-Media-Server/assets/127934645/21abd8f7-34d0-46b7-9641-02419c477ca3)
![image](https://github.com/ant-media/Ant-Media-Server/assets/127934645/c033c2fd-b486-41b4-aeef-7ba9ce299f0a)

**Reply by tremzo** [reply]
@FranciscoNetoBr I was able to start a live stream and played it back successfully using rtmp://129.148.27.211/LiveApp/Z4Ub86CA2QptgLoE1698735366640.

**Reply by FranciscoNetoBr** [reply]
I tested it on three different machines and was unable to connect.

**Reply by muratugureminoglu** [reply]
Hi @FranciscoNetoBr 

I also was able to publish by using the following address. Please check your network, firewall etc. 

rtmp://129.148.27.211/LiveApp/Z4Ub86CA2QptgLoE1698735366640

![Screenshot at 2023-11-01 10-41-09](https://github.com/ant-media/Ant-Media-Server/assets/6337605/082273c9-fe9d-46e7-98c5-f9140ffae209)

**Reply by FranciscoNetoBr** [reply]
I DID IT, I DID IT... THANK YOU, THANK YOU!!!!

YOU GUYS ARE FUCKING!! KISSES!!!!

PS: I will check what happened and find out the reason for the failures.

1 - The processing was high. - But it was not just that. I have to see.

**Reply by MrInternetGitHub** [reply]
what was the issue and how did you fix it?

**Reply by filip-ams** [reply]
Greetings @MrInternetGitHub,

There were two issues in this thread I believe. 

The first one was resolved by freeing up resources on the server (The stream won't be able to start if the cpu usage is too high).

The second issue was related to security rules for the server, you can read up on which ports are needed for Ant Media Server to work properly, you can read up a bit more here:

https://antmedia.io/docs/guides/installing-on-linux/installing-ams-on-linux/#server-ports
